dialog/file.go

```diff
@@ -473,24 +471,30 @@ func (f *fileDialog) refreshDir(dir fyne.ListableURI) {
 }
 
 func (f *fileDialog) setLocation(dir fyne.URI) error {
-	if f.selectedID > -1 {
-		f.files.Unselect(f.selectedID)
-	}
 	if dir == nil {
 		return errors.New("failed to open nil directory")
 	}
+
+	if f.selectedID > -1 {
+		f.files.Unselect(f.selectedID)
+	}
+
 	list, err := storage.ListerForURI(dir)
 	if err != nil {
 		return err
 	}
 
+	isDir, err := storage.CanList(dir)
```

Comment:
I don't know if it's related to the failures but this refactor seems to change behaviour.

This particular line won't add anything because the previous line got a lister, which will error if it cannot list.

I think this code came form inside a for loop so I'm not clear if it wasn't needed or if it was moved to the wrong location.

Reply:
Good point about it not being relevant due to the line above. I don't think it should be needed in the for-loop. We can probably assume that paths higher up in the file system tree are listable and if they are not, we will get an error when we try to change to them. Given that we want to show all of the path up to where we are, I think that is fair.

But yeah, this is not related to the test failures. I think I am handling the volume name incorrectly.